linera-indexer/lib/src/db/postgres/mod.rs

```diff
@@ -114,16 +120,22 @@ impl PostgresDatabase {
         tx: &mut Transaction<'_, Postgres>,
         blob_id: &BlobId,
         data: &[u8],
+        block_hash: Option<CryptoHash>,
+        transaction_index: Option<u32>,
     ) -> Result<(), PostgresError> {
         let blob_id_str = blob_id.hash.to_string();
         let blob_type = format!("{:?}", blob_id.blob_type);
+        let block_hash_str = block_hash.as_ref().map(|h| h.to_string());
+        let transaction_index_i64 = transaction_index.map(|i| i as i64);
 
-        // For now, we don't have block_hash and application_id context here
-        // These could be passed as optional parameters in the future
-        sqlx::query("INSERT INTO blobs (hash, blob_type, data) VALUES ($1, $2, $3) ON CONFLICT (hash) DO NOTHING")
+        sqlx::query(
+            "INSERT INTO blobs (hash, blob_type, data, block_hash, transaction_index) VALUES ($1, $2, $3, $4, $5) ON CONFLICT (hash) DO NOTHING"
```

Comment:
(long line)

Reply:
Fixed in [Various small changes from the review](https://github.com/linera-io/linera-protocol/pull/4988/commits/a75c4c2e0242c276a99b893f1724bde1ade06a70)